src/imitation/contrib/data_loader_wrappers.py

```diff
@@ -0,0 +1,211 @@
+import abc
+import warnings
+from typing import Any, Generic, Iterable, Iterator, List, Mapping, Tuple, TypeVar
+
+import gym
+import numpy as np
+import torch as th
+
+from imitation.data import types
+
+S = TypeVar("S")
+T = TypeVar("T")
+
+
+class DataLoaderWrapper(Generic[S, T], Iterable[T], abc.ABC):
+    def __init__(self, data_loader: Iterable[S]):
+        self.data_loader = data_loader
+
+    @abc.abstractmethod
+    def __iter__(self) -> Iterator[T]:
+        pass
+
+
+class BidirectionalTransform(abc.ABC):
+    @abc.abstractmethod
```

Comment:
If they are pure functions, why are they instance methods? Alternatives:
  - Static methods
  - NamedTuple of functions
(You could be sneaky and keep state with either of these, but you couldn't do it unintentionally, whereas if I'm writing a class it's quite natural to use `self` to store things.)

Reply:
Ah they aren't actually pure. It's more like they are pure with respect to initialization variables. For example, we something like a `CameraObservationScalingWrapper.__init__(self, scale_factor)` which multiplies a key inside dict obs by `scale_factor`.

I'll figure out how to express this better in the docs.

Reply:
I see, so really we're saying two calls to a given *instance* with the same variables should return the same output? No mutable state, in a way.

Reply:
Yes.

I'll probably write in the docs that this ABC is recommended for wrappers with that property.